langchain4j-core/src/test/java/dev/langchain4j/model/chat/mock/ChatModelMock.java

```diff
@@ -46,6 +58,28 @@ public Response<AiMessage> generate(List<ChatMessage> messages) {
         return Response.from(AiMessage.from(staticResponse));
     }
 
+    @Override
+    public Response<AiMessage> generate(List<ChatMessage> messages,
+                                        List<ToolSpecification> toolSpecifications) {
+        requests.add(new ArrayList<>(messages));
+
+        if (exception != null) {
+            throw exception;
+        }
+
+        if (toolExecutionRequest == null) {
```

Comment:
```suggestion
        if (toolExecutionRequest != null) {
```
I guess it was supposed to be `!= null`?